lua/harpoon/cmd-ui.lua

```diff
@@ -0,0 +1,148 @@
+local harpoon = require("harpoon")
+local popup = require("popup")
+local log = require("harpoon.dev").log
+local term = require("harpoon.term")
+
+local M = {}
+
+Harpoon_cmd_win_id = nil
+Harpoon_cmd_bufh = nil
+
+local function close_menu(force_save)
+    force_save = force_save or false
+    local global_config = harpoon.get_global_settings()
+
+    if global_config.save_on_toggle or force_save then
+        require("harpoon.cmd-ui").on_menu_save()
+    end
+
+    vim.api.nvim_win_close(Harpoon_cmd_win_id, true)
+
+    Harpoon_cmd_win_id = nil
+    Harpoon_cmd_bufh = nil
+end
+
+local function create_window()
+    log.trace("_create_window()")
```

Comment:
I am a bit worried about so much duplicated logic.  I just don't want to create bugs that are hard to squuuuuoooosh